canvas_modules/common-canvas/src/common-properties/controls/checkboxset/checkboxset.jsx

```diff
@@ -80,21 +85,49 @@ class CheckboxsetControl extends React.Component {
 		}
 		const checkboxes = [];
 		for (var i = 0; i < this.props.control.values.length; i++) {
+			const tooltipId = uuid4() + "-tooltip-" + this.props.control.name;
+			let tooltip = "";
+			if (this.props.control.description && !(this.props.state === STATES.DISABLED || this.props.state === STATES.HIDDEN) && !this.props.tableControl) {
+				tooltip = (
+					<span >{this.props.control.valueDescs[i]}</span>
+				);
+				// If tooltip has a link, add propertyId in the link object
+				if (this.props.control.description.link) {
+					this.props.control.description.link.propertyId = this.props.propertyId;
+				}
+			}
+			const tooltipIcon = isEmpty(tooltip) ? "" : (
+				<Tooltip
+					id={tooltipId}
+					tip={tooltip}
+					link={this.props.control.description.link ? this.props.control.description.link : null}
+					tooltipLinkHandler={this.props.controller.getHandlers().tooltipLinkHandler}
+					direction="right"
+					className="properties-tooltips"
+					showToolTipOnClick
+				>
+					<Icon type={CARBON_ICONS.INFORMATION} className="properties-control-description-icon-info" />
+				</Tooltip>
+			);
+
 			const id = {
 				name: this.props.propertyId.name,
 				row: i
 			};
 			const val = this.props.control.values[i];
 			const checked = (controlValue.indexOf(val) >= 0);
 			const disabled = this.props.state === STATES.DISABLED || !this.props.controlOpts.values.includes(val);
-			checkboxes.push(<Checkbox
-				disabled={disabled}
-				id={ControlUtils.getControlId(id, this.uuid)}
-				key={val + i}
-				labelText={this.props.control.valueLabels[i]}
-				onChange={this.handleChange.bind(this, val)}
-				checked={checked}
-			/>);
+			checkboxes.push(<div className="wrapper" key={val + i}>
```

Comment:
wrapper isn't a good classname.  That is very generic.

Reply:
Updated with `.checkbox-tooltip-container`